payjoin/src/send/v2/mod.rs

```diff
@@ -110,10 +129,35 @@ impl<'a> SenderBuilder<'a> {
         self,
         min_fee_rate: FeeRate,
     ) -> Result<Sender, BuildSenderError> {
-        Ok(Sender {
-            v1: self.0.build_non_incentivizing(min_fee_rate)?,
+        let pj_url = self.v1.uri.extras.endpoint.clone();
+        let sender = Sender {
+            v1: self.v1.build_non_incentivizing(min_fee_rate)?,
             reply_key: HpkeKeyPair::gen_keypair().0,
-        })
+        };
+        self.persister
+            .save(pj_url.into(), sender.clone())
+            .map_err(|e| InternalBuildSenderError::FailedToPersistSender(ErrorBox::new(e)))?;
+        Ok(sender)
```

Comment:
The 3x duplicate code should also raise flags. Instead I imagine, like the receiver, the SenderBuilder will produce a `NewSender` which implements `persist() -> Result<Sender>` after which a Sender can be loaded with `Sender::load`. This separates the concerns of instantiation and persistence and solves the save / load asymmetry.